tests/publiccloud/storage_perf.pm

```diff
@@ -81,7 +81,7 @@ sub run {
     $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $reg_code,                timeout => 600) if (get_required_var('FLAVOR') =~ m/BYOS/);
     $instance->run_ssh_command(cmd => 'sudo zypper --gpg-auto-import-keys -q in -y fio', timeout => 600);
 
-    my $block_device = '/dev/' . $instance->run_ssh_command(cmd => 'lsblk|grep ' . $disk_size . '|cut -f 1 -d " "');
+    my $block_device = '/dev/' . $instance->run_ssh_command(cmd => 'lsblk -n -l --output NAME,MOUNTPOINT | sort | tail -n1');
```

Comment:
are we sure that last disk is the one we have attached and not the one that Azure temporary disk?

Reply:
there is no evidence which I can give except VRs for all public cloud providers . But after thinking about all possible ways to fix this I picked up this one. it is fragile I understand this but I failed to find bullet prove one they are all fragile. But I can say that it is better than what we have now with the size .  Simply change size of input drive to any random value is no go for me because of 2 reasons : 
1. I don't like when code dictates me what input values I can use for the test 
2. The way how Azure deciding about this temp drive size is black magic for us so "look for size" approach is fragile in the same way as this one :)

Reply:
we agreed about use of openQA variable but later in a chat with @cfconrad  I realized that this way would be super tricky because this temp drive for example is provided **not** for all machine types in Azure which means that we will need to define very smart way to calculate value for this variable which would keep failing every time Azure will decide to add/remove this drive to one or another machine type

Reply:
I understand that and I agree, my question is if we are sure that if we have `sda`, `sdb` and `sdc`, how do we know the last one `sdc` is the right one and not `sdb`? If you say you have doublechecked this yourself, then I trust you and I'm ok with it.

Reply:
I agree with @jlausuch, but I think it was just suffient. We also had the idea to search for a sdX which just don't have any mount point with this:
```
comm <(lsblk -n -l --output NAME,MOUNTPOINT | grep -E '^\s*\S+\s+\S+\s*$' | cut -c 1-3 | sort | uniq)      <(lsblk -n -l --output NAME,MOUNTPOINT | grep -E '^\s*\S+\s*$' | cut -c 1-3 | sort | uniq)      -13
```
but this is quite heavy if the other works :)
I mean the best would be to specify some UUID or other label. But I didn't found anything in terraform doc.

Reply:
ok, fair enough. It's an improvement already :)